pkg/controllers/dynakube/otelc/statefulset/reconciler.go

```diff
@@ -127,24 +128,42 @@ func (r *Reconciler) buildTemplateAnnotations(ctx context.Context) (map[string]s
 		templateAnnotations = r.dk.Spec.Templates.OpenTelemetryCollector.Annotations
 	}
 
+	tlsSecretHash, err := r.calculateSecretHash(ctx, r.dk.ExtensionsTLSSecretName())
+	if err != nil {
+		return nil, err
+	}
+
+	templateAnnotations[api.AnnotationExtensionsSecretHash] = tlsSecretHash
+
+	if r.dk.TelemetryService().IsEnabled() && r.dk.TelemetryService().Spec.TlsRefName != "" {
+		tlsSecretHash, err = r.calculateSecretHash(ctx, r.dk.TelemetryService().Spec.TlsRefName)
```

Comment:
You must calculate hash based on Secret.Data IMO, not secret name.

Reply:
hash IS based on secret.data
```
	tlsSecret, err := query.Get(ctx, types.NamespacedName{...
	tlsSecretHash, err := hasher.GenerateHash(tlsSecret.Data)
```
works for me (collector has been restarted)
```
internal.operator.dynatrace.com/telemetry-service-secret-hash: "778158509" 
internal.operator.dynatrace.com/telemetry-service-secret-hash: "1270306686" 
```